addons/mrp/models/mrp_production.py

```diff
@@ -659,16 +659,18 @@ def _onchange_producing(self):
     @api.onchange('lot_producing_id')
     def _onchange_lot_producing(self):
         if self.product_id.tracking == 'serial':
-            if self.env['stock.move.line'].search_count([
-                ('company_id', '=', self.company_id.id),
-                ('product_id', '=', self.product_id.id),
-                ('lot_id', '=', self.lot_producing_id.id),
-                ('state', '!=', 'cancel')
-            ]):
+            quants = self.env['stock.quant'].search([('product_id', '=', self.product_id.id),
```

Comment:
I would (as you did in scrap.py of mrp) only trigger the search if lot_producing_id is set. I remember you can, via inventory adjustment, create quants without SN for tracked product